idunn/utils/result_filter.py

```diff
@@ -135,19 +135,23 @@ def check(
         if not any(word_matches(q_word, l_word) for l_word in full_label):
             logger.info(
                 "Removed `%s` from results because queried `%s` does not match the result",
-                name,
+                names[0],
                 q_word,
             )
             return False
 
-    # Check if at least 2/3 of the result's name is covered by the query
+    # Check if at least 2/3 of the result is covered by the query in one of these cases:
+    #   - over one of the result names
+    #   - over one of the result names together with one of the admin names
+    #   - over one of the result names together with a postcode
     def coverage(terms):
         return sum(
             any(word_matches(query_word, term) for query_word in query_words) for term in terms
         ) / len(terms)
 
     check_coverage = any(
         coverage(terms) >= (2 / 3)
+        for name_words in map(words, names)
         for terms in [
```

Comment:
Looking at the Eiffel Tower, in the worst cases there will be ~50 translations, ~10 admins which leads to about 1500 calls to `coverage` (which is about quadratic in the number of words of the query, that shouldn't be too expensive).

I don't think it will be an issue but if it is, it could be relevant to take another approach, like checking only a few languages (like French + English + `lang` + local name)

Reply:
A simple way to attenuate this overhead would be to use a lru_cache on `word_matches`, as common words are likely to appear in different languages:

```diff
+@lru_cache(maxsize=10000)
 def word_matches(query_word, label_word):
     """
     Check if two words match with an high degree of confidence.
```

Reply:
Well an extra dict lookup is almost free so I'll include this change. I don't think this will become an issue anyway